File: 3785-find-the-number-of-copy-arrays/find-the-number-of-copy-arrays.cpp
class Solution {
public:
    int countArrays(vector<int>& original, vector<vector<int>>& bounds) {
        int n = original.size();
        int left = bounds[0][0];
        int right = bounds[0][1];

        for (size_t i = 1; i < n; ++i) {
            int diff = original[i] - original[i - 1];

            left = std::max(left + diff, bounds[i][0]);
            right = std::min(right + diff, bounds[i][1]);

            if (left > right)
                return 0;
        }

        return right - left + 1;
    }
};

File: 3785-find-the-number-of-copy-arrays/find-the-number-of-copy-arrays.py
class Solution:
    def countArrays(self, original: list[int], bounds: list[list[int]]) -> int:
        n = len(original)
        left, right = bounds[0]

        for i in range(1, n):
            diff = original[i] - original[i - 1]

            left = max(left + diff, bounds[i][0])
            right = min(right + diff, bounds[i][1])

            if left > right:
                return 0

        return right - left + 1
